src/qt/radiobut.cpp

```diff
@@ -42,6 +44,15 @@ bool wxRadioButton::Create( wxWindow *parent,
     m_qtRadioButton = new QRadioButton( parent->GetHandle() );
     m_qtRadioButton->setText( wxQtConvertString( label ));
 
+    if ( style & wxRB_GROUP )
```

Comment:
To answer the question: if `wxRB_GROUP` is not specified, the first radio button in the window still starts its own group (this is the reason for `wxRB_SINGLE` which can be used to prevent this from happening).

I.e. the logic here should be

    if !HasStyle(wxRB_SINGLE):
        if HasStyle(wxRB_GROUP) or !TryJoiningExistingGroup():
            CreateAndJoinNewGroup()

Reply:
> To answer the question: if `wxRB_GROUP` is not specified, the first radio button in the window still starts its own group (this is the reason for `wxRB_SINGLE` which can be used to prevent this from happening).

@vadz I don't think you are right, at least when it comes to `wxRB_SINGLE`, and here is why:
- consecutive radio buttons that are not explicitly part of a group will still behave as if they were in one (each one will exclude the others when chosen);
- consequence 1: if the first radio buttons in a window do not have any of the 2 styles, then there is no need to explicitly add them to a group because the implicit behavior is the same (and IIRC the same happens in MSW implementation);
- consequence 2: if `wxRB_SINGLE` buttons would not create their own groups in which they would be the sole members, then 2 or more consecutive `wxRB_SINGLE` buttons will not actually be "single".

Reply:
Oops, yes, my logic above was wrong, thanks for correcting. We do need to create a group for `wxRB_SINGLE`, so it should actually be

    if HasStyle(wxRB_SINGLE) || HasStyle(wxRB_GROUP) || !TryJoiningExistingGroup():

But I still think we do need to create a new group for a button without any styles if it can't join an existing one because non-consecutive buttons without any explicit styles are still supposed to be part of the same group from wx API contract point of view.

Reply:
> But I still think we do need to create a new group for a button without any styles if it can't join an existing one because non-consecutive buttons without any explicit styles are still supposed to be part of the same group from wx API contract point of view.

I can't argue about the intention of the API, but IIRC it doesn't work that way on MSW. Take the following case:
Radio 1 [no style]
Radio 2 [no style]
Radio 3 [wxRB_SINGLE]
Radio 4 [no style]
Radio 5 [no style]
Radio 1 & 2 will behave as being in a group together, while 4 & 5 will behave as being in their own group, separate from 1 & 2.

Reply:
Yes, this is correct. And my logic above should result in exactly this behaviour because `TryJoiningExistingGroup()` would return false for the button 1 (as there is nothing to join) and button 4 (as it will encounter a single radio button 3 and also return false).

Reply:
Then I didn't understand what you meant by this:

> non-consecutive buttons without any explicit styles are still supposed to be part of the same group from wx API contract point of view.

But anyway, my point was that buttons 1 & 2 do not _need to_ be in an explicit group, but there's no harm in creating one for them too, of course.

Reply:
> Then I didn't understand what you meant by this:
> 
> > non-consecutive buttons without any explicit styles are still supposed to be part of the same group from wx API contract point of view.

Ah, I see how it was confusing. By "non-consecutive" I meant that they could be intermingled with other controls, e.g. there could be a radio button, text control, static text and another radio button. I did **not** mean that there could be radio buttons with `wxRB_SINGLE` between them.

> But anyway, my point was that buttons 1 & 2 do not _need to_ be in an explicit group, but there's no harm in creating one for them too, of course.

If they don't need an explicit group even if they're not strictly consecutive, then we should skip creating it. Note that creating it has another small advantage: looking for the group would stop as soon as the previous radio button is found instead of having to continue to the very beginning.